definitions/features/foreman_proxy.rb

```diff
@@ -1,21 +1,58 @@
 class Features::ForemanProxy < ForemanMaintain::Feature
   metadata do
     label :foreman_proxy
+
+    confine do
+      file_exists?('/etc/dhcp/dhcpd.conf') &&
+        ForemanMaintain::Utils::SmartProxySSL.instance.ssl_certs_exists?
+    end
   end
 
-  attr_reader :dhcpd_conf_file, :cert_path
+  attr_reader :dhcpd_conf_file, :cert_path, :base_url,
+              :default_ssl_cert_options
 
   def initialize
     @dhcpd_conf_file = '/etc/dhcp/dhcpd.conf'
```

Comment:
@swapab ,
I don't think we require new `DHCP` utility class.
Instead I would prefer to modify `DhcpReservation` to `DHCP` as those APIs are nothing but categorized as `DHCP` APIs under `SmartProxy`.

There are different API categories under smart-proxy. 
Please check [URL](http://projects.theforeman.org/projects/smart-proxy/wiki/API) where smart-proxy APIs are listed, I hope this pretty much helpful.

Lets discuss about these topics in scrum meeting to make code better and clean.